### DISCUSSION TITLE
wget: can't handle reloc type 0x2a in lib '/lib/libc.so.6'

### DISCUSSION THREAD

**Discussion by reimerp** [Freetz-NG/freetz-ng#748] [Allgemein]
Hallo, ich habe mit freetz auf einer 6660 einige Probleme.

wget-gnu und curl verweigern den start mit

> /usr/bin/wget-gnu: can't handle reloc type 0x2a in lib '/lib/libc.so.6'

andere Pakete, die ich integriert habe funktionieren problemlos.

Mein Verdacht fällt auf ssl ?

Ich verwende pfichtner-freetz mit ubuntu 20.04 zum bauen.

**Comment by freetz-user** [comment]
Ich würde mehr auf libusb1 gehen. Denn auf der 7530AX gibt es denn fehler can't handle reloc type 0xa0. Und den kann man umgehen, in dem man Libusb1 Downgrade auf die Version davor oder auf die neue wartet.

**Comment by reimerp** [comment]
libusb ist hier scheinbar unbeteiligt. ldd zeigt:
```
ldd ./build/modified/filesystem/usr/bin/wget-gnu
	linux-gate.so.1 (0xf7fb6000)
	libssl.so.3 => not found
	libcrypto.so.3 => not found
	libc.so.0 => not found
```
ausgeführt in Buildsystem, daher kommt "not found". Auf der FB gibs ja kein ldd

**Comment by fda77** [comment]
@reimerp: auf dem host finde ich `readelf -a DATEI| grep -E "NEED|PATH"` besser. Ich denk das ist mehr ein problem beim setzen des loaders durch patchelf. mach mal ein "distclean" und bau komplett neu. wenn das nicht hilft ist es ein problem mit den puma7, davon hatte ich noch nie ne box

**Comment by reimerp** [comment]
Zumindest funkt wget wenn gegen gnutls gebaut wird.

**Comment by fda77** [comment]
Tja, keine Ahnung woran das liegt. Alle anderen binaries funktionieren? Es könnte die uclib für x86 sein, die wird nur von docsis fritzen verwendet.
Wäre gut wenn sonst jemand mit so Geräten sagen könnte ob wget je funktioonierte

**Comment by reimerp** [comment]
Ich hab ein relativ altes freetz-ng-19016M-194580d4a auf einer 6490 mit wget-gnu (+ssl) das hat noch funktioniert, ist aber openssl 1.0
```
 0x00000001 (NEEDED)                     Gemeinsame Bibliothek [libssl.so.1.0.0]
 0x00000001 (NEEDED)                     Gemeinsame Bibliothek [libcrypto.so.1.0.0]
 0x00000001 (NEEDED)                     Gemeinsame Bibliothek [libc.so.0]
```
Das neue gegen gnutls läuft auch:
```
 0x00000001 (NEEDED)                     Gemeinsame Bibliothek [libnettle.so.8]
 0x00000001 (NEEDED)                     Gemeinsame Bibliothek [libgnutls.so.30]
 0x00000001 (NEEDED)                     Gemeinsame Bibliothek [libc.so.0]
```
Nur das neu gebaute curl (wo die gnutls im menuconfig nicht möglich ist) hat jetzt noch das Issue (genauso wie wget wenn gegen openssl gelinkt)

```
 0x00000001 (NEEDED)                     Gemeinsame Bibliothek [libcurl.so.4]
 0x00000001 (NEEDED)                     Gemeinsame Bibliothek [libssl.so.3]
 0x00000001 (NEEDED)                     Gemeinsame Bibliothek [libcrypto.so.3]
 0x00000001 (NEEDED)                     Gemeinsame Bibliothek [libatomic.so.1]
 0x00000001 (NEEDED)                     Gemeinsame Bibliothek [libc.so.0]
```
Andere Programme (socat, tcpdump, smbd) laufen
Aber socat wiederum hat auch den reloc 2a:
```
 0x00000001 (NEEDED)                     Gemeinsame Bibliothek [libssl.so.3]
 0x00000001 (NEEDED)                     Gemeinsame Bibliothek [libcrypto.so.3]
 0x00000001 (NEEDED)                     Gemeinsame Bibliothek [libc.so.0]
```
zufällig spukt hier auch libssl.so.3 drin rum

Ist jetzt nicht mega wichtig, gnutls geht ja.

**Comment by fda77** [comment]
Openssl 1.0 bekommt seit 3 Jahren keine updates mehr. Daher bitte die Scheisse die Samma Master mit seinen 100en Accounts immer verzapft schnell wieder vergessen, er hat krasse versändnisprobleme! Diesen ganzen Quatsch mit openssl veranstaltet er nur da er es nicht schafft sein oscam statisch zu bauen
Du kannst ja openssl 1.1 versuchen, wenn openssl3 einen Fehler haben sollte, diesen einfach dort melden

Ansonsten wären ja noch mindestens die uclib version, kernel version, docsis version und gar die cpu zwischen 6490 und  6660 nicht gleich

> Nur das neu gebaute curl (wo die gnutls im menuconfig nicht möglich ist)

Könnte man bestimmt machen, wenn man jemanden findet der es macht. Am schnellsten geht wie immer: selbst machen
